src/vpn-api/worker.ts

```diff
@@ -143,13 +155,69 @@ const worker = (instanceId: number) => {
 			),
 		)
 		.tap(() => openvpn.start())
+		.delay(1000)
+		.tap(() => {
+			return vpn.connect({ port: mgtPort, shellPrompt: '' }).then(() => {
+				vpn.connection.shellPrompt = '';
+				// monitor new client connections and map cid to uuid
+				vpn.on('log', data => {
+					if (data.includes('CLIENT:ESTABLISHED,')) {
+						const clientId = data.split(',')[1].trim();
+						const idMapper = (logData: string) => {
+							if (logData.includes('CLIENT:ENV,common_name=')) {
+								const uuid = logData.split('=')[1].trim();
+								// expire any previous mappings for this device
+								if (uuidMap[uuid] != null) {
+									delete cidMap[uuidMap[uuid]];
+									delete uuidMap[uuid];
+								}
+								// register current mapping
+								cidMap[clientId] = uuid;
+								uuidMap[uuid] = clientId;
+								logger.info(
+									`Parsed connect event for client_id=${clientId} uuid=${uuid}`,
+								);
+								vpn.removeListener('log', idMapper);
+							}
+						};
+						vpn.on('log', idMapper);
+					}
+				});
+				// process bytecount events to track realtime data usage
+				vpn.on('data', data => {
+					if (data.bytecount_cli == null) {
+						return;
+					}
+					const clientId = data.bytecount_cli[0];
+					const uuid = cidMap[clientId];
+
+					if (uuid == null) {
+						logger.error(`Unknown CID(${clientId}) from OpenVPN!`);
+						return;
+					}
+
+					const ts = Date.now();
+					const rxBytes = parseInt(data.bytecount_cli[1], 10);
+					const txBytes = parseInt(data.bytecount_cli[2], 10);
+					// send bytecount data to master process
+					if (process.send != null) {
+						process.send({ uuid, ts, rxBytes, txBytes });
```

Comment:
This would be replaced with:

```
metrics.histogram('vpn_session_rx_bytes', rxBytes);
metrics.histogram('vpn_session_tx_bytes', txBytes);
```

Reply:
(thus obviating the call to `Date.now()` and the check for `process.send != null`)